### DISCUSSION TITLE
Cannot use self hosted server

### DISCUSSION THREAD

**Discussion by llitkr** [rustdesk/rustdesk#1772] [Q&A]
Hello. I installed RustDesk today and trying to use this with self-hosted server.

I'm using windows and I followed all of guide.

I installed NSSM, then extracted server files at Program Files/RustDesk Server, Then run hbbr and hbbs with command with my personal domain which points my IP address.

`nssm install "RustDesk hbbs service" "C:\Program Files\RustDesk Server\hbbs.exe" -r mydomain.com -k _
nssm install "RustDesk hbbr Service" "C:\Program Files\RustDesk Server\hbbr.exe" -k _`

Then I also copied the content of pub file placed in same folder of hbbs file and change the RustDesk client file name to "rustdesk-host=mydomain.com,key=[public key]"

Finally I checked at "About RustDesk" popup that my Public key anhd Host name is displayed well.

However program said "Not ready, Please check your connection".

I tried to set my router option to use DMZ(open all port to specific internal IP address) to my Desktop which installed Rustdesk but still it didn't work.

I deleted hbbr, hbbs services and change the command from "mydomain.com" to "mydomain.com:1234" for change the port, but windows file name doesn't allow to use ":" charactor so I just put it in the program option like "mydomain.com:1234" but still cannot connect.

I restarted hbbr, hbbs serviced many times but still cannot connect. I think it's not a problem about port.

What is problem? How can I fix it?

Thank you for read.

**Comment by GhostNaix** [comment]
Can you ensure that the the server ports are not blocked by windows firewall?

**Comment by pedjas** [comment]
Are you connection to server from public internet or from local network?

**Reply by JakeLT22** [reply]
Hi. I'm having issues with the Windows Firewall itself: the only way to make it work is to disable it. The firewall of the cloud server is setup TCP ports 21115 to 21119 and 8000 and UDP port 21116 and working fine.

My connection is from the public internet...

Thank you! :)
